src/GraphQl/Action/EntrypointAction.php

```diff
@@ -117,4 +114,45 @@ private function parseRequest(Request $request): array
 
         return [$query, $operation, $variables];
     }
+
+    private function parseInput(?string $query, ?array $variables, ?string $operation, string $jsonContent): array
+    {
+        $input = json_decode($jsonContent, true);
+
+        if (isset($input['query'])) {
+            $query = $input['query'];
+        }
+
+        if (isset($input['variables'])) {
+            $variables = \is_array($input['variables']) ? $input['variables'] : json_decode($input['variables'], true);
+        }
+
+        if (isset($input['operation'])) {
+            $operation = $input['operation'];
+        }
+
+        return [$query, $operation, $variables];
+    }
+
+    private function applyMapToVariables(Request $request, array $variables): array
+    {
+        $mapValues = json_decode($request->request->get('map'), true);
+        if (!$mapValues) {
+            return $variables;
+        }
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
```

Comment:
You should use the service instead.
IMO using this is overkill.

Reply:
You are right